libs/process/stub.go

```diff
@@ -0,0 +1,167 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+var stubKey int
+
+// WithStub creates process stub for fast and flexible testing of subprocesses
+//
+//	ctx := context.Background()
+//	ctx, stub := WithStub(ctx)
+//	stub.WithDefaultOutput("meeee")
+//	out, err := Background(ctx, []string{"/usr/local/bin/meeecho", "1", "--foo", "bar"})
+//	require.NoError(t, err)
+//	require.Equal(t, "meeee", out)
+//	require.Equal(t, 1, stub.Len())
+//	require.Equal(t, []string{"meeecho 1 --foo bar"}, stub.Commands())
+func WithStub(ctx context.Context) (context.Context, *processStub) {
+	stub := &processStub{responses: map[string]reponseStub{}}
+	ctx = context.WithValue(ctx, &stubKey, stub)
+	return ctx, stub
+}
+
+func runCmd(ctx context.Context, cmd *exec.Cmd) error {
+	stub, ok := ctx.Value(&stubKey).(*processStub)
+	if ok {
+		return stub.run(cmd)
+	}
+	return cmd.Run()
+}
+
+type reponseStub struct {
+	stdout string
+	stderr string
+	err    error
+}
+
+type processStub struct {
+	calls           []*exec.Cmd
+	defaultOutput   string
+	defaultFailure  error
+	defaultCallback func(*exec.Cmd) error
+	responses       map[string]reponseStub
+}
+
+func (s *processStub) WithDefaultOutput(output string) *processStub {
+	s.defaultOutput = output
+	return s
+}
+
+func (s *processStub) WithDefaultFailure(err error) *processStub {
+	s.defaultFailure = err
+	return s
+}
+
+func (s *processStub) WithDefaultCallback(cb func(cmd *exec.Cmd) error) *processStub {
+	s.defaultCallback = cb
+	return s
+}
+
+func (s *processStub) WithStdoutFor(norm, out string) *processStub {
```

Comment:
Can use some comments; i.e. what is `norm`. Had to read everything else to grok it.